python/lsst/daf/butler/script/createRepo.py

```diff
@@ -31,6 +32,8 @@ def createRepo(repo, seed_config=None, standalone=False, override=False, outfile
         URI to the location to create the repo.
     seed_config : `str` or `None`
         Path to an existing YAML config file to apply (on top of defaults).
+    dimension_config : `str` or `None`
```

Comment:
I think this should also be able to take a `Config`.

Reply:
This is the CLI-level API, it only works with strings, so it does not make much sense to allow other types here.

Reply:
I realize this is pre-existing, and so probably out of scope, but I don't see that this extra level of indirection adds any value.  Surely the two lines of implementation here qualify as the kind of "minimal" that we'd permit in the actual Click subcommand definition, right?  May be a question for @n8pease, actually.

Reply:
Sometimes the implementations are quite long and sometimes they are very short. The original motivation was to simplify testing by being able to call the script/ version from test files (which is what the convention is for non-click command lines) without going through forking a subprocess. Of course now we are using the click test runner thing so that is no longer relevant.